### PR TITLE
Add support for Rails 7.0.

### PR DESCRIPTION
**Pull request by aclemons**

### CODE REVIEW

File: lib/active_record/mysql/enum/mysql_adapter.rb [hunk 1]
```diff
@@ -45,20 +45,30 @@ def type_to_sql(type, limit: nil, **_options) # :nodoc:
...
+          private
 
-        private
+          def initialize_type_map(m = type_map)
+            super
+
+            m.register_type(/enum/i) do |sql_type|
+              limit = sql_type.sub(/^enum\('(.+)'\)/i, '\1').split("','").map { |v| v.to_sym }
+              ActiveRecord::Type::Enum.new(limit: limit)
+            end
```
Comment (jebentier) [inline lib/active_record/mysql/enum/mysql_adapter.rb]: Wow, I had to do a little digging into the Rails 7 changes to understand why the below additions were necessary. They've removed all flexibility of new types being bolted onto the adapters by loading the mapping at abstract load time. Thank you for taking care of this and adding the support. Could we DRY up the `register_type` chunk that I've highlighted in this comment into its own helper method? That way we have one way of registering the enum type.

File: lib/active_record/mysql/enum/mysql_adapter.rb [hunk 2]
```diff
@@ -45,20 +45,30 @@ def type_to_sql(type, limit: nil, **_options) # :nodoc:
           end
         end
 
+        if Gem::Version.new(Rails.version.to_f.to_s) < Gem::Version.new('7.0')
```
Comment (jebentier) [inline lib/active_record/mysql/enum/mysql_adapter.rb]: `Rails.version` already outputs a string, so the `to_f.to_s` is unnecessary here, would you mind removing this?

File: lib/active_record/mysql/enum/mysql_adapter.rb [hunk 3]
```diff
@@ -45,20 +45,30 @@ def type_to_sql(type, limit: nil, **_options) # :nodoc:
...
+            end
+          end
+        end
+      end
 
-        def initialize_type_map(m = type_map)
-          super
+      ActiveRecordMysqlAdapter.prepend ActiveRecord::Mysql::Enum::MysqlAdapter
 
+      unless Gem::Version.new(Rails.version.to_f.to_s) < Gem::Version.new('7.0')
```
Comment (jebentier) [inline lib/active_record/mysql/enum/mysql_adapter.rb]: Same request here around removing the `to_f.to_s` from the `Rails.version`

**Review by jebentier — CHANGES_REQUESTED** [review]

File: lib/active_record/mysql/enum/mysql_adapter.rb (hunk 2 above)
Comment (aclemons) [reply lib/active_record/mysql/enum/mysql_adapter.rb]: Sorry about this - I was testing this code back when it was Rails 7.0.0.alpha1 and it was breaking my check here:

```ruby
Gem::Version.new("7.0.0.alpha1") < Gem::Version.new('7.0')
=> true
```

so I was forcing it to a single decimal digit and then back to a string for Gem::Version.

```
irb(main):008:0> "7.0.0.alpha1".to_f.to_s
=> "7.0"
irb(main):009:0> Gem::Version.new("7.0.0.alpha1".to_f.to_s) < Gem::Version.new('7.0')
=> false
```

Anyway, not necessary now and should have removed this before opening. I have removed it now.

File: lib/active_record/mysql/enum/mysql_adapter.rb (hunk 3 above)
Comment (aclemons) [reply lib/active_record/mysql/enum/mysql_adapter.rb]: Done

File: lib/active_record/mysql/enum/mysql_adapter.rb (hunk 1 above)
Comment (aclemons) [reply lib/active_record/mysql/enum/mysql_adapter.rb]: Good call. I think this should be done now on my branch too.

**Review by jebentier — APPROVED** [review]

**Comment by jebentier** [comment]
Awesome. Thanks for taking care of all this. I'll get this merged and staged for a minor version release shortly.

**Pull request merged**